### PR TITLE
Issue 58 standardize clock

### PR DESCRIPTION
**Pull request by qinjunj**
Made changes to use Relative Clock. Works fine only with ILLIXR issue-58-standardize-clock branch.

### CODE REVIEW

**Comment by charmoniumQ** [comment]
This is connected to [ILLIXR #250](https://github.com/ILLIXR/ILLIXR/pull/250). They should be reviewed together, but this should be merged in to OpenVINS first, then assigned a new tag (e.g. "3.1"), then ILLIXR #250 needs to be updated to refer to the new tag in every `config/*.yaml`, and then ILLIXR #250 can be merged.

File: ov_msckf/src/slam2.cpp [hunk 1]
```diff
@@ -183,15 +188,16 @@ class slam2 : public plugin {
...
-		_m_pose.put(_m_pose.allocate(
-			std::chrono::time_point<std::chrono::system_clock>{},
-			Eigen::Vector3f{0, 0, 0},
-			Eigen::Quaternionf{1, 0, 0, 0}
-		));
+		// _m_pose.put(_m_pose.allocate(
+		// 	_m_rtc->now(), 
+		// 	Eigen::Vector3f{0, 0, 0},
+		// 	Eigen::Quaternionf{1, 0, 0, 0}
+		// ));
```
Comment (mhuzai) [inline ov_msckf/src/slam2.cpp]: Why is this commented out?

What's surprising is that this branch still works. In that case, why was this required in the first place?

File: ov_msckf/src/slam2.cpp [hunk 2]
```diff
@@ -183,15 +188,16 @@ class slam2 : public plugin {
 		, sb{pb->lookup_impl<switchboard>()}
 		, _m_pose{sb->get_writer<pose_type>("slow_pose")}
 		, _m_imu_integrator_input{sb->get_writer<imu_integrator_input>("imu_integrator_input")}
-		, _m_begin{std::chrono::system_clock::now()}
+		, _m_rtc{pb->lookup_impl<RelativeClock>()}
```
Comment (mhuzai) [inline ov_msckf/src/slam2.cpp]: Why is this required? It's never used.

File: ov_msckf/src/slam2.cpp [hunk 3]
```diff
@@ -183,15 +188,16 @@ class slam2 : public plugin {
...
+		// , _m_begin{_m_rtc->now()}
```
Comment (mhuzai) [inline ov_msckf/src/slam2.cpp]: Delete.

File: ov_msckf/src/slam2.cpp [hunk 4]
```diff
@@ -215,19 +221,16 @@ class slam2 : public plugin {
...
-		previous_timestamp = timestamp_in_seconds;
-
 		// Feed the IMU measurement. There should always be IMU data in each call to feed_imu_cam
 		assert((datum->img0.has_value() && datum->img1.has_value()) || (!datum->img0.has_value() && !datum->img1.has_value()));
-		open_vins_estimator.feed_measurement_imu(timestamp_in_seconds, (datum->angular_v).cast<double>(), (datum->linear_a).cast<double>());
+		open_vins_estimator.feed_measurement_imu(duration2double(datum->time.time_since_epoch()), datum->angular_v.cast<double>(), datum->linear_a.cast<double>());
+
+		// std::cout << std::fixed << "Time of IMU/CAM: " << timestamp_in_seconds * 1e9 << " Lin a: " << 
+		// 	datum->angular_v[0] << ", " << datum->angular_v[1] << ", " << datum->angular_v[2] << ", " <<
+		// 	datum->linear_a[0] << ", " << datum->linear_a[1] << ", " << datum->linear_a[2] << std::endl;
```
Comment (mhuzai) [inline ov_msckf/src/slam2.cpp]: Remove.

File: ov_msckf/src/slam2.cpp [hunk 5]
```diff
@@ -320,14 +322,14 @@ class slam2 : public plugin {
 	const std::shared_ptr<switchboard> sb;
 	switchboard::writer<pose_type> _m_pose;
     switchboard::writer<imu_integrator_input> _m_imu_integrator_input;
-	time_type _m_begin;
+	std::shared_ptr<RelativeClock> _m_rtc;
+	// time_point _m_begin;
```
Comment (mhuzai) [inline ov_msckf/src/slam2.cpp]: Delete.

File: ov_msckf/src/slam2.cpp (hunk 1 above)
Comment (qinjunj) [reply ov_msckf/src/slam2.cpp]: I think it's because `_m_rtc->now()` might get called before the clock starts. Not sure about the second one. But seeing [this](https://github.com/ILLIXR/ILLIXR/blob/issue-58-standardize-clock/pose_prediction/plugin.cpp#L62), I guess it's not required?

File: ov_msckf/src/slam2.cpp (hunk 2 above)
Comment (qinjunj) [reply ov_msckf/src/slam2.cpp]: It's used by the first dummy pose which is then commented out. If we deem the first dummy pose as not required, this is not required either.

File: ov_msckf/src/slam2.cpp (hunk 1 above)
Comment (mhuzai) [reply ov_msckf/src/slam2.cpp]: Yeah, exactly, pose predict handles the lack of a pose. Also, we do something different [in Kimera](https://github.com/ILLIXR/Kimera-VIO/blob/master/examples/plugin.cpp#L52). We should just not publish anything in VIO's constructor.

Also, technically, the empty pose constructor on `data_format.hpp:157` should return an identity quaternion, not a zero quaternion.

File: ov_msckf/src/slam2.cpp [hunk 6]
```diff
@@ -14,6 +14,7 @@
 #include "common/switchboard.hpp"
 #include "common/data_format.hpp"
 #include "common/phonebook.hpp"
+#include "common/relative_clock.hpp"
```
Comment (mhuzai) [inline ov_msckf/src/slam2.cpp:17]: No longer required.

File: ov_msckf/src/core/VioManager.cpp [hunk 7]
```diff
@@ -567,9 +567,12 @@ void VioManager::do_feature_propagate_update(double timestamp) {
     total_filter_time += (time_total - time_track);
     total_frame_time += time_total;
 
+#ifndef NDEBUG
```
Comment (mhuzai) [inline ov_msckf/src/core/VioManager.cpp:570]: I know these changes don't belong in this PR, but we're short on time and this change is something that we've been meaning to make for some time.

**Review by mhuzai — APPROVED** [review]
